go/mysql/binlog_event_common.go

```diff
@@ -50,10 +50,21 @@ import (
 //	http://dev.mysql.com/doc/internals/en/event-header-fields.html
 type binlogEvent []byte
 
+const (
+	// Default length of the fixed header for v4 events.
+	binlogFixedHeaderLen = 19
+	// The offset from 0 where the type is stored as 1 byte.
+	binlogEventTypeOffset = 4
+	// Offset from 0 where the 4 byte length is stored.
+	binlogEventLenOffset = 9
+	// Byte length of the checksum suffix.
+	binlogChecksumLen = 4
+)
+
 // dataBytes returns the event bytes without header prefix and without checksum suffix
 func (ev binlogEvent) dataBytes(f BinlogFormat) []byte {
 	data := ev.Bytes()[f.HeaderLength:]
-	data = data[0 : len(data)-4]
+	data = data[0 : len(data)-binlogChecksumLen]
```

Comment:
We should really check `BinlogFormat` to see which checksum is being used, before assuming a checksum is at all used. Sample code:

```go
	switch f.ChecksumAlgorithm {
	case mysql.BinlogChecksumAlgCRC32:
                .... size is binlogChecksumLen
	case mysql.BinlogChecksumAlgOff:
		... size is 0
	default:
		return vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "unsupported checksum algorithm: %v", format.ChecksumAlgorithm)
```

Reply:
I'd rather not expand this PR. I don't disagree with you that we can improve this but it's not directly related and I want to keep this focused.